docs/DeveloperGuide.adoc

```diff
@@ -218,29 +218,53 @@ image::CommitActivityDiagram.png[]
 
 ===== Aspect: How undo & redo executes
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Saves the entire CardiBuddy.
 ** Pros: Easy to implement.
 ** Cons: May have performance issues in terms of memory usage.
 * **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
+** Pros: Will use less memory (e.g. for `delete`, just save the deck being deleted).
+** Cons: We must ensure that the implementation of each individual command is correct.
 
 ===== Aspect: Data structure to support the undo/redo commands
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
+* **Alternative 1 (current choice):** Use a list to store the history of CardiBuddy states.
 ** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
+** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedCardiBuddy`.
 * **Alternative 2:** Use `HistoryManager` for undo/redo
 ** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+=== Search and Filter feature
+Users can choose to find a `Deck` or a `Flashcard` through:
 
-_{Explain here how the data encryption feature will be implemented}_
+* searching by keywords from the title of a `Deck` or from the `Question` or `Answer` from a `Flashcard`.
+* filtering by tags from the `Tag(s)` associated with a `Deck` or a `Flashcard`
 
-// end::dataencryption[]
+Both `search` and `filter` have the and-or functionality where adding `&` between keywords returns decks and flashcards
+with both terms before and after the `&` symbol.
+For example, `filter Y1 & hard` will return decks and flashcards that are tagged with both `Y1` and `hard`.
+
+The following sequence diagram shows how the `Search` command works:
+
+image::SearchFilterDiagram.png[]
```

Comment:
sc:SearchCommand was created but there's no contructor?

Suggestion: Why is getFilteredDeckList.size() called when filtered list has already been returned to sc?

Suggestion: sdkp is passed into sc, but sc does not call any of its methods?